WcaOnRails/app/models/user.rb

```diff
@@ -293,6 +283,30 @@ def after_confirmation
     end
   end
 
+  def software_team?
+    team_member?('software')
+  end
+
+  def results_team?
+    team_member?('results')
+  end
+
+  def wrc_team?
+    team_member?('wrc')
+  end
+
+  def wdc_team?
+    team_member?('wdc')
+  end
+
+  def team_member?(team)
+    self.team_members.where(team_id: self.teams.find_by_friendly_id(team)).any?(&:current_member?)
```

Comment:
I looked into why this works without the `.id`, and doesn't work with the `.id`. When you add the `.id`, a lot of our tests fail while trying to look up teams that don't exist. For example:

```
  82) UsersController editing user data after registering for a competition user cannot change name
     Failure/Error: patch :update, id: user.id, user: { name: "Johnny 5" }
     NoMethodError:
       undefined method `id' for nil:NilClass
     # ./app/models/user.rb:303:in `team_member?'
     # ./app/models/user.rb:287:in `software_team?'
     # ./app/models/user.rb:311:in `admin?'
     # ./app/models/user.rb:319:in `can_edit_users?'
     # ./app/controllers/users_controller.rb:103:in `redirect_if_cannot_edit_user'
     # ./app/controllers/users_controller.rb:54:in `update'
     # ./spec/controllers/users_controller_spec.rb:133:in `block (4 levels) in <top (required)>'
     # -e:1:in `<main>'

```

Right now, this method will return false if the given team just doesn't exist. I'd rather have protection against typos in `team_friendly_id`, so I believe we should change both `team_member?` and `team_leader?` to use `self.teams.find_by_friendly_id!`.

We'll then need to update all our tests to actually create the teams they need. As discussed a while back, I believe that sould go in a `config.before` block in [spec/support/database_cleaner.rb](https://github.com/jfly/worldcubeassociation.org/blob/d96aa516bd46876528debaa335bc03c932dbff0b/WcaOnRails/spec/support/database_cleaner.rb), which would allow you to remove the team creation stuff from the users factory where it currently is